src/stream-tcp.c

```diff
@@ -908,14 +910,54 @@ static int StreamTcpPacketStateNone(ThreadVars *tv, Packet *p,
         PacketQueueNoLock *pq)
 {
     if (p->tcph->th_flags & TH_RST) {
-        StreamTcpSetEvent(p, STREAM_RST_BUT_NO_SESSION);
-        SCLogDebug("RST packet received, no session setup");
-        return -1;
+        if (stream_config.midstream == FALSE) {
+            StreamTcpSetEvent(p, STREAM_RST_BUT_NO_SESSION);
+            SCLogDebug("RST packet received, no session setup");
+            return -1;
+        }
+        ssn = StreamTcpNewSession(p, stt->ssn_pool_id);
```

Comment:
these changes should be in a separate commit, as they are changing Suricata's midstream behavior and are not just about flags logging

Reply:
you'll have to convince me that this makes sense at all btw, it seems setting up a tcp session based on a RST packet makes no sense

Reply:
The use case here is in anomaly detection; in sudden increase in flows containing only FIN or RST packets would be an indication of strange TCP behaviour, and possibly something that would need further indication.

I could probably make it configurable.